### PR TITLE
Refactor view controllers

### PR DESCRIPTION
**Pull request by kristen**
Here is my refactorization(don't think that's a real word) of the main view controllers. Please make comments if you think there is a better way for me to do something, I wont be offended and will be happy to learn from you guys!

Also, I don't know if you guys knew this, but swift has a didSet hook that allows you to do things on the objects right after they have been set. I think it is clean to add the delegates for our tableView, gridView and mapView here because it cleans up the viewDidLoad() and moves the code to a related place. I thought it was pretty cool when I learned this.

### CODE REVIEW

File: Mapstagram/Controllers/MediaViewController.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+//  MediaViewController.swift
+//  Mapstagram
+//
+//  Created by Kristen on 3/7/15.
+//  Copyright (c) 2015 JMK. All rights reserved.
+//
+
+import UIKit
+
+class MediaViewController: UIViewController {
```
Comment (jamietsao) [inline Mapstagram/Controllers/MediaViewController.swift]: I really like this common base controller.  Nice reuse of code, and a perfect case for inheritance!

File: Mapstagram/GridViewController.swift [hunk 2]
```diff
@@ -8,66 +8,38 @@
...
-    // refresh control
-    var refreshControl: UIRefreshControl!
-    
-    @IBOutlet weak var gridCollectionView: UICollectionView!
-    
-    // array of media to display
-    var mediaArray: [Media] = []
+class GridViewController: MediaViewController {
+    
+    @IBOutlet weak var gridCollectionView: UICollectionView! {
```
Comment (jamietsao) [inline Mapstagram/GridViewController.swift]: This is a really cool pattern indeed.  Nice!

**Comment by jamietsao** [comment]
Looks good to me.  Merge it!

**Comment by jamietsao** [comment]
Actually, do you mind moving all the Grid/Map/View files into their own directories to match the Groups as part of this code refactoring?  BTW, I've typically been creating "Models", "Views", "Controllers" groups, but I'm totally open to other ways.

**Comment by kristen** [comment]
@jamietsao I will move the files into the folders and groups you suggested right now.

**Comment by kristen** [comment]
@jamietsao I just finished rearranging the files. I've been using ViewControllers instead of Controllers, so I did that here. You can merge this in for me if you think it is good.

**Comment by jamietsao** [comment]
Looks good.  Merging.

**Pull request merged**